File: SuperiorPackGroup/BLL/AuditTrailReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class AuditTrailReportBLL

    Private m_AuditTrailReportTableAdapter As AuditTrailReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As AuditTrailReportTableAdapter
        Get
            If (Me.m_AuditTrailReportTableAdapter Is Nothing) Then
                Me.m_AuditTrailReportTableAdapter = New AuditTrailReportTableAdapter
                Me.m_AuditTrailReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_AuditTrailReportTableAdapter
        End Get
    End Property

    Public Function GetAuditTrailReport(ByVal fromDate As Nullable(Of Date), ByVal toDate As Nullable(Of Date)) As SPGReports.AuditTrailReportDataTable

        Return Adapter.GetAuditTrailReport(fromDate, toDate)

    End Function

End Class

File: SuperiorPackGroup/BLL/AuditTrailReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class AuditTrailReportBLL
	{

		private AuditTrailReportTableAdapter m_AuditTrailReportTableAdapter = null;

		public AuditTrailReportTableAdapter Adapter
		{
			get
			{
				if (m_AuditTrailReportTableAdapter == null)
				{
					m_AuditTrailReportTableAdapter = new AuditTrailReportTableAdapter();
					m_AuditTrailReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_AuditTrailReportTableAdapter;
			}
		}

		public SPGReports.AuditTrailReportDataTable GetAuditTrailReport(DateTime? fromDate, DateTime? toDate)
		{

			return Adapter.GetAuditTrailReport(fromDate, toDate);

		}

	}

}